assembl/static2/js/app/components/debate/thread/threadView.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { Grid } from 'react-bootstrap';
+
+import Permissions, { connectedUserCan } from '../../../utils/permissions';
+import TopPostFormContainer from '../../../components/debate/common/topPostFormContainer';
+import Tree from '../../../components/common/tree';
+import Post, { PostFolded } from '../../../components/debate/thread/post';
+import InfiniteSeparator from '../../../components/common/infiniteSeparator';
+
+class ThreadView extends React.Component {
+  render() {
+    const {
+      isUserConnected,
+      idea,
+      contentLocaleMapping,
+      refetchIdea,
+      lang,
+      noRowsRenderer,
+      getTopPosts,
+      initialRowIndex
+    } = this.props;
+    return (
+      <div className="overflow-x">
+        {!isUserConnected || connectedUserCan(Permissions.ADD_POST)
+          ? <TopPostFormContainer ideaId={idea.id} refetchIdea={refetchIdea} isColumnViewInline={false} messageColumns={[]} />
```

Comment:
You shouldn't pass the `messageColumns` here at all. The component should init with a check to see if it has the prop, if it doesn't, init it's own state to `[]`, if it needs to. The internal state of the `TopPostFormContainer` should be it's own concern. The parent component is concerned with passing the *appropriate* data through.